src/Plugins/ComplexCore/test/RawBinaryReaderTest.cpp

```diff
@@ -0,0 +1,148 @@
+
+  // Write binary file
+  std::array<char, 10> buffer = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+  std::fstream myFile(inputFile, std::ios::out | std::ios::binary);
+  myFile.write(buffer.data(), buffer.size());
+  myFile.close();
+
+  // Instantiate the filter, a DataStructure object and an Arguments Object
+  RawBinaryReaderFilter filter;
+  DataStructure ds;
+  Arguments args;
+
+  // Create default Parameters for the filter.
+  args.insertOrAssign(RawBinaryReaderFilter::k_InputFile_Key, std::make_any<FileSystemPathParameter::ValueType>(fs::path(inputFile)));
+  args.insertOrAssign(RawBinaryReaderFilter::k_ScalarType_Key, std::make_any<NumericType>(NumericType::uint8));
+  args.insertOrAssign(RawBinaryReaderFilter::k_NumberOfComponents_Key, std::make_any<int32>(1));
+  args.insertOrAssign(RawBinaryReaderFilter::k_Endian_Key, std::make_any<ChoicesParameter::ValueType>(0));
+  args.insertOrAssign(RawBinaryReaderFilter::k_SkipHeaderBytes_Key, std::make_any<uint64>(0));
+  args.insertOrAssign(RawBinaryReaderFilter::k_CreatedAttributeArrayPath_Key, std::make_any<DataPath>(createdAttributeArrayPathValue));
+
+  // Preflight the filter and check result
+  auto preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.valid());
+
+  // Execute the filter and check the result
+  auto executeResult = filter.execute(ds, args);
+  REQUIRE(executeResult.result.valid());
+
+  // Check the results
+  auto* resultArray = ds.getDataAs<DataArray<uint8>>(createdAttributeArrayPathValue);
+  for(int i = 0; i < 10; i++)
+  {
+    uint8 resultVal = resultArray->at(i);
+    REQUIRE(resultVal == i);
+  }
+}
+
+TEST_CASE("Core::RawBinaryReaderFilter: Invalid filter execution")
+{
+  std::string arrayName = "Imported Array";
+  DataPath createdAttributeArrayPathValue = DataPath(std::vector<std::string>{arrayName});
+
+  // Instantiate the filter, a DataStructure object and an Arguments Object
+  RawBinaryReaderFilter filter;
+  DataStructure ds;
+  Arguments args;
+
+  // Test empty input file
+  std::string inputFile;
+
+  args.insertOrAssign(RawBinaryReaderFilter::k_InputFile_Key, std::make_any<FileSystemPathParameter::ValueType>(fs::path(inputFile)));
+  args.insertOrAssign(RawBinaryReaderFilter::k_ScalarType_Key, std::make_any<NumericType>(NumericType::uint8));
+  args.insertOrAssign(RawBinaryReaderFilter::k_NumberOfComponents_Key, std::make_any<int32>(1));
+  args.insertOrAssign(RawBinaryReaderFilter::k_Endian_Key, std::make_any<ChoicesParameter::ValueType>(0));
+  args.insertOrAssign(RawBinaryReaderFilter::k_SkipHeaderBytes_Key, std::make_any<uint64>(0));
+  args.insertOrAssign(RawBinaryReaderFilter::k_CreatedAttributeArrayPath_Key, std::make_any<DataPath>(createdAttributeArrayPathValue));
+
+  auto preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.invalid());
+
+  // Test non-existent input file
+  inputFile = "/tmp/this_does_not_exist.bin";
```

Comment:
Use the "unit_test_dirs.h" to use the suitable test directory. This will not work on Windows computers. Line 115 has the proper way to do this but that uses the fs::temp_directory_path() which oddly may fail on some systems. So use the ones in the unit_test_dirs.hpp file.